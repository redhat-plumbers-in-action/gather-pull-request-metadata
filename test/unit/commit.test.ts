import { describe, it, expect, beforeEach } from 'vitest';

import {
  ICommitTestContext,
  commitContextFixture,
} from './fixtures/commit.fixture';

describe('Commit Object', () => {
  beforeEach<ICommitTestContext>(context => {
    context.commits = commitContextFixture.commits;
  });

  it<ICommitTestContext>('can be instantiated', context =>
    context.commits.map(commitItem => expect(commitItem).toBeDefined()));

  it<ICommitTestContext>('getTitle()', context =>
    context.commits.map(commitItem => {
      expect(
        commitItem.getTitle(commitItem.message.body)
      ).length.lessThanOrEqual(74);

      expect(commitItem.getTitle(commitItem.message.body)).toMatchSnapshot();
    }));

  it<ICommitTestContext>('getCherryPicks()', context =>
    context.commits.map(commitItem => {
      expect(
        commitItem.getCherryPicks(commitItem.message.body)
      ).toMatchSnapshot();
    }));
});
