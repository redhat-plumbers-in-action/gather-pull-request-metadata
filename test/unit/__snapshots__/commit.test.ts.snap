// Vitest Snapshot v1

exports[`Commit Object > getCherryPicks() 1`] = `[]`;

exports[`Commit Object > getCherryPicks() 2`] = `
[
  {
    "sha": "12345",
  },
  {
    "sha": "1234567890123456789012345678901234567890",
  },
]
`;

exports[`Commit Object > getTitle() 1`] = `"message-1"`;

exports[`Commit Object > getTitle() 2`] = `"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa ..."`;
